packages/@react-spectrum/dialog/docs/Dialog.mdx

```diff
@@ -37,58 +44,317 @@ import {Heading, Text} from '@react-spectrum/typography';
+A standard Dialog has the following anatomy:
+
+<ExampleImage>
+  <DialogAnatomy />
+</ExampleImage>
+
+Each of these sections can be populated by providing the following components to your Dialog as children:
+`Header`, `Heading` (title), `Divider`, `Content` (body), `ButtonGroup`, and `Footer`.
 
-### Accessibility
 
-*If the component has accessibility needs, include this section.*
+### Examples
+A typical Dialog with a title, contents, and action buttons can be created like so:
+
+```tsx example
+<DialogTrigger>
+  <ActionButton>Publish</ActionButton>
+  {(close) => (
+    <Dialog>
+      <Heading>Publish 3 pages</Heading>
+      <Divider />
+      <Content>Confirm publish?</Content>
+      <ButtonGroup>
+        <Button variant="secondary" onPress={close}>Cancel</Button>
+        <Button variant="cta" onPress={close} autoFocus>Confirm</Button>
+      </ButtonGroup>
+    </Dialog>
+  )}
+</DialogTrigger>
+```
+
+A dismissable Dialog forgoes a footer section and its associated action buttons in favor
+of rendering a close button. If your Dialog is dismissable, make sure that you do not include a footer section or
+a ButtonGroup. While the Dialog will hide any elements occupying the footer section, it will not remove them from the DOM, 
+resulting in possible tab order issues.
+
+```tsx example
+<DialogTrigger isDismissable>
+  <ActionButton>Status</ActionButton> 
+  <Dialog>
+    <Heading>Status</Heading>
+    <Divider />
+    <Content>Printer Status: Connected</Content>
+  </Dialog>
+</DialogTrigger>
+```
+
+It is important to note that the `Heading`, `Header`, `Content`, and `Footer` content elements accept any renderable 
+node, not just strings. This allows you to create Dialogs for more complex workflows, such as including a form
+for the user to fill out or adding confirmation checkboxes.
+
+```tsx example
+<DialogTrigger>
+  <ActionButton>Register</ActionButton> 
+  {(close) => (
+    <Dialog>
+      <Heading>Register for newsletter</Heading>
+      <Header>
+        <StatusLight variant="negative">
+          Registration Status
+        </StatusLight>
+      </Header>
+      <Divider />
+      <Content>
+        <Form>
+          <TextField label="First Name" placeholder="John" autoFocus />
+          <TextField label="Last Name" placeholder="Smith" />
+          <TextField label="Street Address" placeholder="123 Any Street" />
+          <TextField label="City" placeholder="San Francisco" />
+        </Form>
+      </Content>
+      <Footer>
+        <Checkbox>
+          I want to receive updates for exclusive offers in my area.
+        </Checkbox>
+      </Footer> 
+      <ButtonGroup>
+        <Button variant="secondary" onPress={close}>Cancel</Button>
+        <Button variant="cta" onPress={close}>Register</Button>
+      </ButtonGroup>
+    </Dialog>
+  )}
+</DialogTrigger>
+```
 
-### Internationalization
+The example below illustrates how a Dialog with a hero image could be rendered via the `hero` slot:
```

Comment:
Yep, we moved away from including a table of all available slots since it is more advanced usage and will be covered in a separate doc. The `hero` slot is really the only one that I feel needs a direct shout out here since banner image dialogs are  covered in the spectrum site specifically and is the only slot which we don't have a component for. I think a Hero component might not be a bad idea if other components end up using the same slot name for a similar purpose